src/Controller/CRUDController.php

```diff
@@ -1083,6 +1072,19 @@ public function getRequest()
         return $this->container->get('request_stack')->getCurrentRequest();
     }
 
+    protected function addRenderExtraParams(array $parameters = []): array
```

Comment:
Can you please add type hints

Reply:
Type hints is meaningless here. As far as i see, it is not customary in your project to write meaningless type hints.

Reply:
Indeed, and there is a linter to remove unnecessary typehint

Reply:
Please do add them anyway. `mixed[]` allows us to know we looked into giving precise phpdoc but just couldn't. It allows us to know due diligence has been done. Also, if we ever add stricter static analysis to this project (and I hope we will someday) it will ask for these type hints. Let's do it from the start please. Less work for us.

Reply:
@greg0ire in my opinion `mixed[]` type is equivalent to `array` and therefore is meaningless. Among other things, this type is perceived as an indexed array (`array<int, mixed>`). However, the function expects an associative array `array<string, mixed>` in the argument. Ignoring the array key type may be critical in the future. Compare examples [one](https://phpstan.org/r/f5e64811-4bb0-42a4-b7ef-fcbb87ac8085) and [two](https://phpstan.org/r/013b61f2-dc4b-449d-bd7b-9320d60a3c28).

This form is supported by Psalm and PHPStan, but not supported by PHPStorm. I find it more appropriate to use `array` type.

I can add a special annotation for the Psalm.

```php
/**
 * @psalm-param array<string, mixed> $parameters
 * @psalm-return array<string, mixed>
 */
protected function addRenderExtraParams(array $parameters = []): array
```

Reply:
> in my opinion mixed[] type is equivalent to array and therefore is meaningless. 

It is equivalent indeed, but `mixed[]` is better than nothing, because it says "we already looked into it, you don't need to. Nothing to be improved here.". That said, if there is something more precise we can use we should `array<int, mixed>`

> but not supported by PHPStorm.

I use vim, with https://github.com/dense-analysis/ale, which means I have phpstan and psalm integration in vim, so I would be all for using `array<string, mixed>`. Also, it means I don't know how annoying this is in PHPStorm. I will let other decide what is best (maybe support in PHPStorm is coming soon?), but I'd like it to be that description with either `@param`, `@return` or `@psalm-param`, `@psalm-return`

Reply:
Phpstan understand this. Phpstorm dont.

Reply:
Uuuuuh yeah that's what I understood the first time, why do you say that?

Reply:
I thought you wasnt sure about it, so I just wanted to confirm it to you. Sorry if it wasn’t necessary

Reply:
No problem, what I'm not sure about is whether this is going to be really annoying for PHPStorm users or if an update is coming soon enough that we can go with `@param` instead of `@psalm-param`.

Reply:
> No problem, what I'm not sure about is whether this is going to be really annoying for PHPStorm users or if an update is coming soon enough that we can go with `@param` instead of `@psalm-param`.

As a Phpstorm user, I sometime use `array<int, string>`. Phpstorm just understand `array`. So I see no impact if we use this in Sonata.

Reply:
Ok then it's settled. @peter-gribanov , please add non prefixed phpdoc